src/App.js

```diff
@@ -1,19 +1,61 @@
-import React from 'react';
-import './App.css';
+import React, { Component } from 'react';
 
+import TodoList from './components/TodoList/TodoList';
+import todos from './api/todos';
+import NewTodo from './components/NewTodo/NewTodo';
 import users from './api/users';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Static list of todos</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+import './App.css';
+
+class App extends Component {
+  state = {
+    todosData: [...todos],
+    usersData: [...users],
+  };
+
+  getTodosWithUsers = (todosList, usersList) => todosList
+    .map(todo => ({
+      ...todo,
+      user: usersList.find(user => user.id === todo.userId),
+    }));
+
+  addTodo = (text, userId) => {
+    const newTodo = this.createTodoItem(text, userId);
+
+    this.setState(prevState => ({
+      todosData: [...prevState.todosData, newTodo],
+    }));
+  };
+
+  createTodoItem(value, userId) {
+    return {
+      title: value,
+      completed: false,
+      userId: Number(userId),
+      id: this.state.todosData.length + 1,
+    };
+  }
+
+  render() {
+    const { todosData, usersData } = this.state;
+    const preparedTodos = this.getTodosWithUsers(todosData, usersData);
```

Comment:
what about Dima's comment about this line?

Reply:
fix